nx/test/nx/defn/grad_test.exs

```diff
@@ -60,6 +60,51 @@ defmodule Nx.Defn.GradTest do
       assert stop_grad_tuple_meta(Nx.tensor(1), Nx.tensor(1)) == {Nx.tensor(1.0), Nx.tensor(1.0)}
     end
 
+    defn concatenate_grad(t) do
+      grad(t, stop_grad(Nx.sum(Nx.sum(Nx.concatenate([t, t]), axes: [0]))))
+    end
+
+    defn concatenate_grad_power(t) do
+      grad(
+        t,
+        stop_grad(Nx.sum(Nx.sum(Nx.concatenate([Nx.power(t, 2), Nx.power(t, 3)]), axes: [0])))
```

Comment:
No need for double sum. Also, please move concatenate tests alongside the tests for reshape, squeeze, etc. They have their own describe IIRC.